torch/csrc/jit/passes/quantization.cpp

```diff
@@ -503,8 +503,56 @@ void InsertQuantDequantNodesForParam(
   }
 }
 
+void InsertQuantDequantNodesForParam(
+    script::Method& method,
+    const std::string& param_name,
+    const std::function<std::tuple<std::string, float, int>(float, float)>&
+        getQParamFunc,
+    at::ScalarType t) {
+  AT_ASSERT(getQParamFunc != nullptr);
+  auto params_to_insert_qdq = getQuantizableParamsofName(method, param_name);
+
+  for (auto& param_info : params_to_insert_qdq) {
+    // Fetch scale for input activation and weight to the quantizable
+    // node for which bias need to be quantized
+    Node* n = param_info.n;
+    // Check if this node has weight attr as input
+    size_t param_index = getParamIndexinOpArgs(n, std::string("weight"));
+    if (param_index >= n->inputs().size()) {
+      // No attribute by name weight
+      continue;
+    }
+    std::vector<size_t> node_inputs_idx{0, param_index};
+    std::array<float, 2> scale_factors = {0, 0};
```

Comment:
Please add some comments here. It's hard to tell why this code is necessary and what exactly it is doing.

Reply:
Done